pandas/core/indexes/datetimes.py

```diff
@@ -1747,7 +1747,33 @@ def freq(self, value):
     is_year_end = _field_accessor(
         'is_year_end',
         'is_year_end',
-        "Logical indicating if last day of year (defined by frequency)")
+        """
+        Return a boolean indicating whether the date is the last day of the
+        year.
+
+        Returns
+        -------
+        is_year_end : Series of boolean.
```

Comment:
Ah, this return type depends on the type of self. So I think

is_year_end : Series or DatetimeIndex
    The same type as the original data. Series will have the same
    name and index. DatetimeIndex will have the same name.